FF8.Fields/Sources/LookupService/Field.FSLookupService.cs

```diff
@@ -0,0 +1,71 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+// ReSharper disable StringLiteralTypo
+
+namespace FF8.Fields
+{
+    public static partial class Field
+    {
+        public sealed class FSLookupService : ILookupService
+        {
+            private readonly String _rootPath;
+
+            public FSLookupService(String rootPath)
+            {
+                if (!Directory.Exists(rootPath))
+                    throw new DirectoryNotFoundException(rootPath);
+
+                _rootPath = rootPath;
+            }
+
+            public IEnumerable<Info> EnumerateAll()
+            {
+                return Enumerate("*.mim");
+            }
+
+            public IEnumerable<Info> Enumerate(String mask)
+            {
+                foreach (String filePath in Directory.EnumerateFiles(_rootPath, mask, SearchOption.AllDirectories))
+                {
+                    String fieldName = Path.GetFileNameWithoutExtension(filePath);
+                    if (SkipInvalidFields(fieldName))
+                        continue;
+
+                    IDataProvider dataProvider = new FsDataProvider(filePath);
+                    yield return new Info(fieldName, dataProvider);
+                }
+            }
+
+            private static Boolean SkipInvalidFields(String fieldName)
+            {
+                switch (fieldName)
```

Comment:
This should not rely on field name. Let's leave it as-is but in the future we should come up with a method to detect corrupted JSM and ignore field on-the-go